src/cargo/sources/git/utils.rs

```diff
@@ -565,7 +565,14 @@ pub fn fetch(repo: &git2::Repository, url: &str,
         let mut opts = git2::FetchOptions::new();
         opts.remote_callbacks(cb)
             .download_tags(git2::AutotagOption::All);
-        try!(remote.fetch(&[refspec], Some(&mut opts), None));
-        Ok(())
+
+        match network::with_retry(cargo_config, ||{
+            remote.fetch(&[refspec], Some(&mut opts), None)
+        }){
+            Ok(_) => Ok(()),
+            Err(e) =>{
+                Err(git2::Error::from_str(&format!("{}", e)))
```

Comment:
The try leads me back to this comment https://github.com/rust-lang/cargo/pull/2396#issuecomment-216349535 since the code that wraps this was changed from a CargoResult to a Result. I think the two options are matching and re-packing or changing the signature of  with_authentication back to a CargoResult. Im not sure why the method was changed but this is what made sense to match the new expected return type.  

Is there another way to do this without the match?